### ISSUE TITLE
Coding ward numbers / names

### ISSUE THREAD

**Issue by GabPoll** [openhealthcare/elcid#129]
Wards are currently free text. However they are also a key sorting criteria on the list. This has the potential to create confusion.
For example T9 jumps above t9, but is trumped by T 10, because of the space.
Instead if we had a look up list, typing in 10, would automatically bring up option "T10", keeping that consistent.
Again, perhaps this may only be an issue for haem, but there the ward round strictly goes by bed order and it's crucial to have the list in the right order. In the future, ITU, Queen sq ITU rounds follow similar pattern.

Of course this is a short term solution before CDR integration but that won't happen for a while.

**Comment by michaeledwardmarks** [comment]
A few extra comments:

- [ ] As part of this we need to also fix the thing where one bit of the location list offers UCH and another UCLH
- [ ]  And personally I would re-set default location to UCLH whenever a patient is added.
It remains the only site where UCH staff use elCID

- [ ] I would strongly consider making the Ward List a drop down over a-lookup

**Comment by michaeledwardmarks** [comment]
- [ ] Last thing is about the way in which the list orders these Wards (regardless of if they are typed correctly). These should be in actual numerical order (e.g T16>T15)

At the moment you get something like
T1
T1n (e.g all the T10,T11 etc)
T2
T3

**Comment by GabPoll** [comment]
@michaeledwardmarks clarification on point above - UCLH is the trust. UCH is the hospital, so all should change to UCH.

Agree with reverse numerical order - i thought i'd written that somewhere, but in essence top of list has to be top of tower - T16, then work down. Currently does the reverse and juniors find it v frustrating.

**Comment by davidmiller** [comment]
Coded ward names implemented and deployed to elcid04-test !

**Comment by GabPoll** [comment]
awesome

**Comment by michaeledwardmarks** [comment]
@davidmiller
This is fantastic

A few issues:
- The lookup list behaves a bit weirdly.
Try typing just "T" - it assumes you want Accident and Emergency - you can't see "T1" "T2" until you start typing the numbers.

- T6/T7/T8/T9 (in fact anything that isn't T1X) don't seem to show up in the list.

- It currently doesn't default to displaying the canonical term
E.g 
AAU = Acute Assessment Unit = T1 = T 1
See ID-Inpatients list where there are two patients both on the same actual ward but one displaying Acute Assessment Unit and one displaying T1

**Comment by GabPoll** [comment]
well spotted @michaeledwardmarks

**Comment by davidmiller** [comment]
@michaeledwardmarks These are great points :) 

When we generate the suggestions for a lookuplist, we take all possible entries that match the currently typed text, and then display them in (mostly) alphabetical order. 
This algorithm works well in the common case. This algorithm does not work well when your entry has only two characters.

In principle, we can change the algorithm.
Pragmatically, we're not going to fix it now.

Not sure quite what you mean by t6 not showing up on the list ? 
What happens when you just type 6 ? 

The lack of synonym support is due to the underlying storage format.
It's the same issue as #214 (I've updated that to reflect this issue)

Basically a lack of perfect foresight early on made some subsequent changes problematical.
Implementing this has many of the same issues as openhealthcare/opal#268 and #647 - we should do all at the same time.

Again, this is not a thing for right now - we need to do a deployment *before* making substantial changes to the storage layer.

Basically: 

Is having the lookuplist as is better than it was beforehand even with these issues?

If so, let's open a new discussion issue for point 1, and prioritize #214 as part of 0.5.
If not, I will back out this change and we can leave the whole thing for 0.5

**Comment by GabPoll** [comment]
I was about to report a bug, but realise it's the 3rd of @michaeledwardmarks points above - the synonyms don't seem to be working for wards. See also tropical list. (by comparison, it's working fine for diagnosis).

I can't reproduce @michaeledwardmarks point "T6/T7/T8/T9 (in fact anything that isn't T1X) don't seem to show up in the list." - works fine for me.

To answer your point @davidmiller about whether this is better than before, and my answer is YES, as long as the synonym issue is easily fixable. The funny looking up is not a big deal - people will quickly type t6 or t16 by convention, overriding the funny lookup of the letter.

**Comment by michaeledwardmarks** [comment]
So the non leading 1, tower wards T6-9
show up if you lead with 
- just the number 
- lower case t

If you type "T6" though they don't show up.
I think probably this is enough of a step forward to do it as is.

**Comment by GabPoll** [comment]
this is done, right?

**Comment by davidmiller** [comment]
Modulo the lack of synonyms, yes!

**Comment by michaeledwardmarks** [comment]
I'm reopening this because the lack of synonym support is an ongoing issue according to real users (thanks to Hannah the ward ID doctor sat next to me).

Specific issues:
- [] Lookup list does not bring up T6/T7/T8 if you start with a T (it does if you start with a 6 for example, but that isn't what real users do)
- [] Lack of synonyms (although in admin) means you still get T8 T 8 issues with ward patient ordering

**Comment by GabPoll** [comment]
What if we made the ward list drop down? Would that be an easy fix?
________________________________

**Comment by michaeledwardmarks** [comment]
Drop down would be an easy solution

**Comment by davidmiller** [comment]
I think we should just do that.

On Sat, 5 Sep 2015 at 19:05 Michael Marks <notifications@github.com> wrote:

> Drop down would be an easy solution
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openhealthcare/elcid/issues/129#issuecomment-137981766>
> .
>
-- 
Best regards

David Miller
Open Health Care

**Comment by michaeledwardmarks** [comment]
-[ ] Make drop down
1) Will this remain editable in admin (that would be sensible in case it turns out we have missed a ward!
2) Assume in the first instance this will be context independent (e.g you won't get a specific drop down depending on which hospital the patient is listed as being at)

**Comment by GabPoll** [comment]
Agree admin editable.
I would make it context independent in first instance. I had thought about that, but thought context dependence would add a layer of complexity to what is essentially a bug fix solution at the first pass.

**Comment by michaeledwardmarks** [comment]
This is good.
We need to shorten the look-up list to make this more managable.I have done this on heroku admin.

My personal feeling is that the major users at the moment are ward docs, so it is slightly frustrating that all the NHNN wards (which are rarely needed) are at the top of the list. Not sure much can be done about that until we look at a contextual drop down based on hospital.


Note for deployment - copy over the Ward List from Heroku instead of the one currently on elCID-live.

**Comment by GabPoll** [comment]
thank you fixing this
